pkg/controller/flinkapplication/flink_state_machine.go

```diff
@@ -77,22 +80,46 @@ func (s *FlinkStateMachine) updateApplicationPhase(ctx context.Context, applicat
 }
 
 func (s *FlinkStateMachine) shouldRollback(ctx context.Context, application *v1alpha1.FlinkApplication) bool {
-	if application.Status.DeployHash == "" {
+	if application.Status.DeployHash == "" && application.Status.LastSeenError == "" {
```

Comment:
Adding this condition doesn't make sense to me — the idea behind this check was that if this is the first deploy we can't roll back because there's nothing to roll back _to_, so we might as well just keep trying.